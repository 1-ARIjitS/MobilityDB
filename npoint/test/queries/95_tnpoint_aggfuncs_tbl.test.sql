-------------------------------------------------------------------------------

SELECT asText(tcentroid(temp)) FROM ( VALUES
	(NULL::tnpoint),
	('Npoint(1, 0.5)@2000-01-01'),
	('{Npoint(1, 0.3)@2000-01-01, Npoint(1, 0.5)@2000-01-02, Npoint(1, 0.5)@2000-01-03}')) t(temp);
SELECT asText(tcentroid(temp)) FROM ( VALUES
	(tnpoint 'Npoint(1, 0.5)@2000-01-01'),
	('{Npoint(1, 0.3)@2000-01-01, Npoint(1, 0.5)@2000-01-02, Npoint(1, 0.5)@2000-01-03}'),
	(NULL)) t(temp);
/* Errors */
SELECT asText(tcentroid(temp)) FROM ( VALUES
	(tnpoint '{Npoint(1, 0.3)@2000-01-01, Npoint(1, 0.5)@2000-01-02, Npoint(1, 0.5)@2000-01-03}'),
	('[Npoint(1, 0.2)@2000-01-01, Npoint(1, 0.4)@2000-01-02, Npoint(1, 0.5)@2000-01-03]')) t(temp);

-------------------------------------------------------------------------------

SELECT numInstants(tcount(inst)) FROM tbl_tnpoint_inst;
SELECT numInstants(wcount(inst, '1 hour')) FROM tbl_tnpoint_inst;
SELECT numInstants(tcentroid(inst)) FROM tbl_tnpoint_inst;
SELECT k%10, numInstants(tcount(inst)) FROM tbl_tnpoint_inst GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numInstants(wcount(inst, '1 hour')) FROM tbl_tnpoint_inst GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numInstants(tcentroid(inst)) FROM tbl_tnpoint_inst GROUP BY k%10 ORDER BY k%10;

SELECT numInstants(tcount(ti)) FROM tbl_tnpoint_instset;
SELECT numInstants(wcount(ti, '1 hour')) FROM tbl_tnpoint_instset;
SELECT numInstants(tcentroid(ti)) FROM tbl_tnpoint_instset;
SELECT k%10, numInstants(tcount(ti)) FROM tbl_tnpoint_instset GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numInstants(tcount(ti)) FROM tbl_tnpoint_instset GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numInstants(tcentroid(ti)) FROM tbl_tnpoint_instset GROUP BY k%10 ORDER BY k%10;

SELECT numSequences(tcount(seq)) FROM tbl_tnpoint_seq;
SELECT numSequences(wcount(seq, '1 hour')) FROM tbl_tnpoint_seq;
SELECT numSequences(tcentroid(seq)) FROM tbl_tnpoint_seq;
SELECT k%10, numSequences(tcount(seq)) FROM tbl_tnpoint_seq GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numSequences(wcount(seq, '1 hour')) FROM tbl_tnpoint_seq GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numSequences(tcentroid(seq)) FROM tbl_tnpoint_seq GROUP BY k%10 ORDER BY k%10;

SELECT numSequences(tcount(ts)) FROM tbl_tnpoint_seqset;
SELECT numSequences(wcount(ts, '1 hour')) FROM tbl_tnpoint_seqset;
SELECT numSequences(tcentroid(ts)) FROM tbl_tnpoint_seqset;
SELECT k%10, numSequences(tcount(ts)) FROM tbl_tnpoint_seqset GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numSequences(wcount(ts, '1 hour'))) FROM tbl_tnpoint_seqset GROUP BY k%10 ORDER BY k%10;
SELECT k%10, numSequences(tcentroid(ts)) FROM tbl_tnpoint_seqset GROUP BY k%10 ORDER BY k%10;

-------------------------------------------------------------------------------
