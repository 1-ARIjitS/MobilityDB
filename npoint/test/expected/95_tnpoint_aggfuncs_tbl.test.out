SELECT asText(tcentroid(temp)) FROM ( VALUES
	(NULL::tnpoint),
	('Npoint(1, 0.5)@2000-01-01'),
	('{Npoint(1, 0.3)@2000-01-01, Npoint(1, 0.5)@2000-01-02, Npoint(1, 0.5)@2000-01-03}')) t(temp);
                                                                                               astext                                                                                               
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {POINT(55.752647998401 78.9538133549325)@2000-01-01 00:00:00+00, POINT(48.7186629128278 77.7640705101509)@2000-01-02 00:00:00+00, POINT(48.7186629128278 77.7640705101509)@2000-01-03 00:00:00+00}
(1 row)

SELECT asText(tcentroid(temp)) FROM ( VALUES
	(tnpoint 'Npoint(1, 0.5)@2000-01-01'),
	('{Npoint(1, 0.3)@2000-01-01, Npoint(1, 0.5)@2000-01-02, Npoint(1, 0.5)@2000-01-03}'),
	(NULL)) t(temp);
                                                                                               astext                                                                                               
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {POINT(55.752647998401 78.9538133549325)@2000-01-01 00:00:00+00, POINT(48.7186629128278 77.7640705101509)@2000-01-02 00:00:00+00, POINT(48.7186629128278 77.7640705101509)@2000-01-03 00:00:00+00}
(1 row)

/* Errors */
SELECT asText(tcentroid(temp)) FROM ( VALUES
	(tnpoint '{Npoint(1, 0.3)@2000-01-01, Npoint(1, 0.5)@2000-01-02, Npoint(1, 0.5)@2000-01-03}'),
	('[Npoint(1, 0.2)@2000-01-01, Npoint(1, 0.4)@2000-01-02, Npoint(1, 0.5)@2000-01-03]')) t(temp);
<<<<<<< HEAD
ERROR:  Cannot aggregate temporal values of different type
SELECT numInstants(tcount(inst)) FROM tbl_tnpoint_inst;
=======
ERROR:  Cannot aggregate temporal values of different duration
SELECT numInstants(tcount(inst)) FROM tbl_tnpointinst;
>>>>>>> Synchronize npoint with latest MobilityDB version
 numinstants 
-------------
         100
(1 row)

<<<<<<< HEAD
SELECT numInstants(wcount(inst, '1 hour')) FROM tbl_tnpoint_inst;
=======
SELECT numInstants(wcount(inst, '1 hour')) FROM tbl_tnpointinst;
>>>>>>> Synchronize npoint with latest MobilityDB version
 numinstants 
-------------
         201
(1 row)

<<<<<<< HEAD
SELECT numInstants(tcentroid(inst)) FROM tbl_tnpoint_inst;
=======
SELECT numInstants(tcentroid(inst)) FROM tbl_tnpointinst;
>>>>>>> Synchronize npoint with latest MobilityDB version
 numinstants 
-------------
         100
(1 row)

<<<<<<< HEAD
SELECT k%10, numInstants(tcount(inst)) FROM tbl_tnpoint_inst GROUP BY k%10 ORDER BY k%10;
=======
SELECT k%10, numInstants(tcount(inst)) FROM tbl_tnpointinst GROUP BY k%10 ORDER BY k%10;
>>>>>>> Synchronize npoint with latest MobilityDB version
 ?column? | numinstants 
----------+-------------
        0 |          10
        1 |          10
        2 |          10
        3 |          10
        4 |          10
        5 |          10
        6 |          10
        7 |          10
        8 |          10
        9 |          10
(10 rows)

<<<<<<< HEAD
SELECT k%10, numInstants(wcount(inst, '1 hour')) FROM tbl_tnpoint_inst GROUP BY k%10 ORDER BY k%10;
=======
SELECT k%10, numInstants(wcount(inst, '1 hour')) FROM tbl_tnpointinst GROUP BY k%10 ORDER BY k%10;
>>>>>>> Synchronize npoint with latest MobilityDB version
 ?column? | numinstants 
----------+-------------
        0 |          20
        1 |          20
        2 |          20
        3 |          20
        4 |          20
        5 |          20
        6 |          20
        7 |          20
        8 |          20
        9 |          20
(10 rows)

<<<<<<< HEAD
SELECT k%10, numInstants(tcentroid(inst)) FROM tbl_tnpoint_inst GROUP BY k%10 ORDER BY k%10;
=======
SELECT k%10, numInstants(tcentroid(inst)) FROM tbl_tnpointinst GROUP BY k%10 ORDER BY k%10;
>>>>>>> Synchronize npoint with latest MobilityDB version
 ?column? | numinstants 
----------+-------------
        0 |          10
        1 |          10
        2 |          10
        3 |          10
        4 |          10
        5 |          10
        6 |          10
        7 |          10
        8 |          10
        9 |          10
(10 rows)

<<<<<<< HEAD
SELECT numInstants(tcount(ti)) FROM tbl_tnpoint_instset;
=======
SELECT numInstants(tcount(ti)) FROM tbl_tnpointi;
>>>>>>> Synchronize npoint with latest MobilityDB version
 numinstants 
-------------
         476
(1 row)

<<<<<<< HEAD
SELECT numInstants(wcount(ti, '1 hour')) FROM tbl_tnpoint_instset;
=======
SELECT numInstants(wcount(ti, '1 hour')) FROM tbl_tnpointi;
>>>>>>> Synchronize npoint with latest MobilityDB version
 numinstants 
-------------
        1328
(1 row)

<<<<<<< HEAD
SELECT numInstants(tcentroid(ti)) FROM tbl_tnpoint_instset;
=======
SELECT numInstants(tcentroid(ti)) FROM tbl_tnpointi;
>>>>>>> Synchronize npoint with latest MobilityDB version
 numinstants 
-------------
         476
(1 row)

<<<<<<< HEAD
SELECT k%10, numInstants(tcount(ti)) FROM tbl_tnpoint_instset GROUP BY k%10 ORDER BY k%10;
=======
SELECT k%10, numInstants(tcount(ti)) FROM tbl_tnpointi GROUP BY k%10 ORDER BY k%10;
>>>>>>> Synchronize npoint with latest MobilityDB version
 ?column? | numinstants 
----------+-------------
        0 |          34
        1 |          64
        2 |          51
        3 |          36
        4 |          44
        5 |          44
        6 |          45
        7 |          68
        8 |          48
        9 |          42
(10 rows)

<<<<<<< HEAD
SELECT k%10, numInstants(tcount(ti)) FROM tbl_tnpoint_instset GROUP BY k%10 ORDER BY k%10;
=======
SELECT k%10, numInstants(tcount(ti)) FROM tbl_tnpointi GROUP BY k%10 ORDER BY k%10;
>>>>>>> Synchronize npoint with latest MobilityDB version
 ?column? | numinstants 
----------+-------------
        0 |          34
        1 |          64
        2 |          51
        3 |          36
        4 |          44
        5 |          44
        6 |          45
        7 |          68
        8 |          48
        9 |          42
(10 rows)

<<<<<<< HEAD
SELECT k%10, numInstants(tcentroid(ti)) FROM tbl_tnpoint_instset GROUP BY k%10 ORDER BY k%10;
=======
SELECT k%10, numInstants(tcentroid(ti)) FROM tbl_tnpointi GROUP BY k%10 ORDER BY k%10;
>>>>>>> Synchronize npoint with latest MobilityDB version
 ?column? | numinstants 
----------+-------------
        0 |          34
        1 |          64
        2 |          51
        3 |          36
        4 |          44
        5 |          44
        6 |          45
        7 |          68
        8 |          48
        9 |          42
(10 rows)

<<<<<<< HEAD
SELECT numSequences(tcount(seq)) FROM tbl_tnpoint_seq;
=======
SELECT numSequences(tcount(seq)) FROM tbl_tnpointseq;
>>>>>>> Synchronize npoint with latest MobilityDB version
 numsequences 
--------------
          100
(1 row)

<<<<<<< HEAD
SELECT numSequences(wcount(seq, '1 hour')) FROM tbl_tnpoint_seq;
=======
SELECT numSequences(wcount(seq, '1 hour')) FROM tbl_tnpointseq;
>>>>>>> Synchronize npoint with latest MobilityDB version
 numsequences 
--------------
          100
(1 row)

<<<<<<< HEAD
SELECT numSequences(tcentroid(seq)) FROM tbl_tnpoint_seq;
=======
SELECT numSequences(tcentroid(seq)) FROM tbl_tnpointseq;
>>>>>>> Synchronize npoint with latest MobilityDB version
 numsequences 
--------------
          100
(1 row)

<<<<<<< HEAD
SELECT k%10, numSequences(tcount(seq)) FROM tbl_tnpoint_seq GROUP BY k%10 ORDER BY k%10;
=======
SELECT k%10, numSequences(tcount(seq)) FROM tbl_tnpointseq GROUP BY k%10 ORDER BY k%10;
>>>>>>> Synchronize npoint with latest MobilityDB version
 ?column? | numsequences 
----------+--------------
        0 |           10
        1 |           10
        2 |           10
        3 |           10
        4 |           10
        5 |           10
        6 |           10
        7 |           10
        8 |           10
        9 |           10
(10 rows)

<<<<<<< HEAD
SELECT k%10, numSequences(wcount(seq, '1 hour')) FROM tbl_tnpoint_seq GROUP BY k%10 ORDER BY k%10;
=======
SELECT k%10, numSequences(wcount(seq, '1 hour')) FROM tbl_tnpointseq GROUP BY k%10 ORDER BY k%10;
>>>>>>> Synchronize npoint with latest MobilityDB version
 ?column? | numsequences 
----------+--------------
        0 |           10
        1 |           10
        2 |           10
        3 |           10
        4 |           10
        5 |           10
        6 |           10
        7 |           10
        8 |           10
        9 |           10
(10 rows)

<<<<<<< HEAD
SELECT k%10, numSequences(tcentroid(seq)) FROM tbl_tnpoint_seq GROUP BY k%10 ORDER BY k%10;
=======
SELECT k%10, numSequences(tcentroid(seq)) FROM tbl_tnpointseq GROUP BY k%10 ORDER BY k%10;
>>>>>>> Synchronize npoint with latest MobilityDB version
 ?column? | numsequences 
----------+--------------
        0 |           10
        1 |           10
        2 |           10
        3 |           10
        4 |           10
        5 |           10
        6 |           10
        7 |           10
        8 |           10
        9 |           10
(10 rows)

<<<<<<< HEAD
SELECT numSequences(tcount(ts)) FROM tbl_tnpoint_seqset;
=======
SELECT numSequences(tcount(ts)) FROM tbl_tnpoints;
>>>>>>> Synchronize npoint with latest MobilityDB version
 numsequences 
--------------
          483
(1 row)

<<<<<<< HEAD
SELECT numSequences(wcount(ts, '1 hour')) FROM tbl_tnpoint_seqset;
=======
SELECT numSequences(wcount(ts, '1 hour')) FROM tbl_tnpoints;
>>>>>>> Synchronize npoint with latest MobilityDB version
 numsequences 
--------------
          502
(1 row)

<<<<<<< HEAD
SELECT numSequences(tcentroid(ts)) FROM tbl_tnpoint_seqset;
=======
SELECT numSequences(tcentroid(ts)) FROM tbl_tnpoints;
>>>>>>> Synchronize npoint with latest MobilityDB version
 numsequences 
--------------
          494
(1 row)

<<<<<<< HEAD
SELECT k%10, numSequences(tcount(ts)) FROM tbl_tnpoint_seqset GROUP BY k%10 ORDER BY k%10;
=======
SELECT k%10, numSequences(tcount(ts)) FROM tbl_tnpoints GROUP BY k%10 ORDER BY k%10;
>>>>>>> Synchronize npoint with latest MobilityDB version
 ?column? | numsequences 
----------+--------------
        0 |           65
        1 |           37
        2 |           51
        3 |           43
        4 |           51
        5 |           34
        6 |           48
        7 |           58
        8 |           57
        9 |           41
(10 rows)

<<<<<<< HEAD
SELECT k%10, numSequences(wcount(ts, '1 hour'))) FROM tbl_tnpoint_seqset GROUP BY k%10 ORDER BY k%10;
ERROR:  syntax error at or near ")"
LINE 1: SELECT k%10, numSequences(wcount(ts, '1 hour'))) FROM tbl_tn...
                                                       ^
SELECT k%10, numSequences(tcentroid(ts)) FROM tbl_tnpoint_seqset GROUP BY k%10 ORDER BY k%10;
=======
SELECT k%10, numSequences(wcount(ts, '1 hour'))) FROM tbl_tnpoints GROUP BY k%10 ORDER BY k%10;
ERROR:  syntax error at or near ")"
LINE 1: SELECT k%10, numSequences(wcount(ts, '1 hour'))) FROM tbl_tn...
                                                       ^
SELECT k%10, numSequences(tcentroid(ts)) FROM tbl_tnpoints GROUP BY k%10 ORDER BY k%10;
>>>>>>> Synchronize npoint with latest MobilityDB version
 ?column? | numsequences 
----------+--------------
        0 |           65
        1 |           37
        2 |           51
        3 |           43
        4 |           51
        5 |           35
        6 |           48
        7 |           58
        8 |           57
        9 |           41
(10 rows)

